### DISCUSSION TITLE
[Typescript][React Native] AsyncStorage types error

### DISCUSSION THREAD

**Discussion by mangjuned** [pmndrs/jotai#1324] [Q&A] [answered]
i got error ```No overload matches this call.``` when adding types to ```atomWithStorage```. Is i'm missing something?

```ts

const storage = createJSONStorage(() => AsyncStorage);

//No overload matches this call.
export const ATOM = atomWithStorage<string>('key','', storage);

```

**Playground Link:** [Provided](https://www.typescriptlang.org/id/play?ssl=1&ssc=23&pln=1&pc=1#code/JYWwDg9gTgLgBAbwL5wGZQiOByAVhGAQ2GwChRJZFCZMB1YGACwGVapCBzAUwBo4AxlG41uAKRYB5AHJtoXbinSYc+IsAD0AVxjAANgGdsAbnLho8AIIGAngDsBcjjzQYs2AALDCAmAFo7GmAAN24-QlsHPwN2BQ0I+wFo2J4TUlIBCDsYuBj5FwBeQW8YcSlZFO4ACiqASjgCgD44a0SnBVrTUg0NaQg4CFCoPQhCABM4EBoBJm4DOGZgeYFCPT0AOlJuAA9KeEzsqwAVSQBZBrgaekZWSoAeGKhgO05GquwAa24bbF5sX9ylU66R6qGe3C2uwsgiyOUsJ1OACYLlcQAxmO0eA8YE8Xm9Pt9fv9gd0NGC7Nw4KQgA)

**Comment by dai-shi** [comment, marked as answer]
`createJSONStorage` isn't type safe. You can pass `any` for a workaround. https://tsplay.dev/NnK2om

You could also create a custom storage with type guards.

**Comment by dai-shi** [comment]
btw, if we don't annotate types, we will get `unknown` atom type, which is correct, and we are required to do runtime type checks.

https://tsplay.dev/WvKL3w

<img width="558" alt="image" src="https://user-images.githubusercontent.com/490574/182311897-f28514e8-49c4-4119-8e02-cc546b0f479a.png">
